File: Bits/getbit.js
/**
 * Récupère un bit d'un nombre selon un index
 * @param {number} number
 * @param {number} position - basé sur 0.
 * @return {number} 0 ou 1
 */
const getBit = (int, position) => (int >> position) & 1;  

// Tests
console.log(getBit(2, 0)); // 0
console.log(getBit(2, 1)); // 1

File: Go/Bits/getBit.go
package main

import . "fmt"

//Récupère le i-ème bit de n.
func getBit(n, i int) int {
	return n >> i & 1
}

func main() {
	Println(getBit(2, 0)) //0
	Println(getBit(2, 1)) //1
}
